### PR TITLE
added correct category to condition with tests

### PR DESCRIPTION
**Pull request by 5t33**
#### What does this PR do?

Adds correct fhir condition.category.

#### Related JIRA tickets:

UPMCFHIR-272

We determined UPMCFHIR-273 was determined unnecessary.

### CODE REVIEW

File: src/main/java/tr/com/srdc/cda2fhir/transform/ResourceTransformerImpl.java [hunk 1]
```diff
@@ -1671,6 +1671,27 @@ else if (!fhirImmunization.getNotGiven()) {
...
+		conditionCategory.setCode("encounter-diagnosis");
+		conditionCategory.setDisplay("Encounter Diagnosis");
+		cond.addCategory().addCoding(conditionCategory);
+		return cond;
+	}
+
+	@Override
+	public Condition tIndication2ConditionProblemListItem(Indication cdaIndication) {
+		Condition cond = tIndication2Condition(cdaIndication);
+		Coding conditionCategory = new Coding().setSystem("http://hl7.org/fhir/condition-category");
```
Comment (mmccall) [inline src/main/java/tr/com/srdc/cda2fhir/transform/ResourceTransformerImpl.java]: Can you add the OID for this code to the tOid2Url function, and call that get the code system?

**Review by mmccall — COMMENTED** [review]
seems pretty good, just 1 change. Can you put the OID in value set transformer for that URL and use that function in both places where you set the URL? It helps to keep it all in the same place. If you do that, this seems good.

**Review by mmccall — CHANGES_REQUESTED** [review]
see previous comment, if you can put the url in oid function, this is good after that.

File: src/main/java/tr/com/srdc/cda2fhir/transform/ResourceTransformerImpl.java (hunk 1 above)
Comment (5t33) [reply src/main/java/tr/com/srdc/cda2fhir/transform/ResourceTransformerImpl.java]: I updated the indication2condition to either be 2 functions: indication2conditionProblemListItem and indication2conditionEncounter... so I can't really map the one indication OID to two different URLs...

**Comment by 5t33** [comment]
@mmccall  Updated. LMK what you think.

**Review by mmccall — APPROVED** [review]

**Pull request merged**